docs/docsite/rst/dev_guide/module_lifecycle.rst

```diff
@@ -1,50 +1,118 @@
+4. When a module or plguin has been deprecated for four release cycles, we remove the code and replace it with a tombstone entry in the routing configuration. Modules and plugins that are removed are no longer shipped with Ansible. The tombstone entry helps users find alternative modules and plugins.
+
+For modules and plugins in collections, the lifecycle is similar. Since ansible-base 2.10, it is no longer possible to mark modules as 'preview' or 'stable'.
 
 .. _deprecating_modules:
 
-Deprecating modules
-===================
+Deprecating modules and plugins in the Ansible main repository
+==============================================================
 
-To deprecate a module, you must:
+To deprecate a module in ansible-core, you must:
 
 1. Rename the file so it starts with an ``_``, for example, rename ``old_cloud.py`` to ``_old_cloud.py``. This keeps the module available and marks it as deprecated on the module index pages.
-2. Mention the deprecation in the relevant ``CHANGELOG``.
-3. Reference the deprecation in the relevant ``porting_guide_x.y.rst``.
+2. Mention the deprecation in the relevant changelog (create a changelog fragment with a section ``deprecated_features``).
+3. Reference the deprecation in the relevant ``porting_guide_core_x.y.rst``.
 4. Add ``deprecated:`` to the documentation with the following sub-values:
 
   :removed_in: A ``string``, such as ``"2.10"``; the version of Ansible where the module will be replaced with a docs-only module stub. Usually current release +4. Mutually exclusive with :removed_by_date:.
-  :remove_by_date: (Added in Ansible 2.10). An ISO 8601 formatted date when the module will be removed. Usually 2 years from the date the module is deprecated. Mutually exclusive with :removed_in:.
+  :remove_by_date: (Added in ansible-base 2.10). An ISO 8601 formatted date when the module will be removed. Usually 2 years from the date the module is deprecated. Mutually exclusive with :removed_in:.
   :why: Optional string that used to detail why this has been removed.
   :alternative: Inform users they should do instead, for example, ``Use M(whatmoduletouseinstead) instead.``.
 
-* note: with the advent of collections and ``routing.yml`` we might soon require another entry in this file to mark the deprecation.
-
 * For an example of documenting deprecation, see this `PR that deprecates multiple modules <https://github.com/ansible/ansible/pull/43781/files>`_.
   Some of the elements in the PR might now be out of date.
 
-Changing a module name
-======================
+Deprecating modules and plugins in a collection
+===============================================
+
+To deprecate a module in a collection, you must:
+
+1. Add a ``deprecation`` entry to ``plugin_routing`` in ``meta/runtime.yml``. For example, to deprecate the module ``old_cloud``, add::
```

Comment:
Sorry to bug, but can you change these to code-blocks? These rst pages get translated and the translation engine can detect and ignore code-blocks but not something like these.

Reply:
I tried to update them. I'm not 100% sure about the indentation of the code-blocks inside the enumerations.